docs: Updates Cloudinary Astro Integration Guides (#9550)
withastro/docs: src/content/docs/en/guides/media/cloudinary.mdx

service: Cloudinary
i18nReady: true
---
import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';

[Cloudinary](https://cloudinary.com) is an image and video platform and headless Digital Asset Manager (DAM) that lets you host assets and deliver them from their content delivery network (CDN).

When delivering from Cloudinary, you additionally get access to their Transformation API, giving you the ability to edit your assets with tools like background removal, dynamic cropping and resizing, and generative AI.

## Using Cloudinary in Astro

Cloudinary supports a wide variety of SDKs that can be used depending on your Astro environment.

> The [Cloudinary Astro SDK](https://astro.cloudinary.dev/) supports native Astro components, including an Image, Video, and Upload component, as well as a Loader that can be used with Astro's Content Layer.

Author: makes a lot of sense. will go with your suggestion, thank you

```jsx
---
import { CldUploadWidget } from 'astro-cloudinary';
---
<CldUploadWidget uploadPreset="<Upload Preset>">
  <button>Upload</button>
</CldUploadWidget>
```

For signed uploads, you can find [a guide and example](https://astro.cloudinary.dev/clduploadwidget/signed-uploads) on the Astro Cloudinary docs.

[Learn more about CldUploadWidget](https://astro.cloudinary.dev/clduploadwidget/basic-usage).

## Loading Images & Videos

The Astro Content Layer can be used to load Cloudinary assets with the `cldAssetsLoader`.

To load images or videos, add:

```js
import { defineCollection } from 'astro:content';
import { cldAssetsLoader } from 'astro-cloudinary/loaders';

export const collections = {
  assets: defineCollection({
    loader: cldAssetsLoader({
      folder: '<Folder>' // Optional, without loads root directory
    })
  }),
}
```

You can then use the [getCollection or getEntry functions](/en/guides/content-collections/#querying-collections) to select one or many images or videos from your collection.

[Learn more about cldAssetsLoader](https://astro.cloudinary.dev/cldassetsloader/basic-usage).

> ## Generating Social Media Cards

Reviewer: ```suggestion
## Generating Cloudinary image URLs
```
I think the as the URL helper is better page structure here, and this becomes one example of where it's useful.

> Social Media Cards are useful for increasing engagement on social posts.
> 
> To create a Social card, generate a Cloudinary URL and utilize Open Graph and Twitter meta tags to embed the image on your website:

Reviewer: ```suggestion
The Astro Cloudinary SDK provides a `getCldOgImageUrl()` helper for generating and using URLs for your images. Use this when you need a URL instead of a component to display your image. 

One common use for a URL is for an Open Graph image in `<meta>` tags for social media cards. This helper, like the components, provides you access to Cloudinary transformations to create dynamic, unique social cards for any of your pages.

The following example shows the necessary `<meta>` tags for a social media card, using `getCldOgImageUrl()` to generate an Open Graph image:
```

> ```jsx

Reviewer: ``````suggestion
```astro title="src/components/SocialMediaCard.astro"
``````

---
import { getCldOgImageUrl } from 'astro-cloudinary/helpers';
const ogImageUrl = getCldOgImageUrl({ src: '<Public ID>' });
---
<meta property="og:image" content={ogImageUrl} />
<meta property="og:image:secure_url" content={ogImageUrl} />
<meta property="og:image:width" content="1200" />
<meta property="og:image:height" content="630" />
<meta property="twitter:title" content="<Twitter Title>" />
<meta property="twitter:card" content="summary_large_image" />
<meta property="twitter:image" content={ogImageUrl} />
```

> The `getCldOgImageUrl` helper utilizes the same API as the CldImage component, meaning you can use Cloudinary transformations to create dynamic, unique social cards for any of your pages.

Reviewer: ```suggestion
```
moved this idea up, where I think it's a nicer intro

> Find [Social Media Card templates](https://astro.cloudinary.dev/templates/social-media-cards) on the Astro Cloudinary docs.

Reviewer: ```suggestion
Find [Cloudinary Social Media Card templates](https://astro.cloudinary.dev/templates/social-media-cards) on the Cloudinary docs.
```
We can't exactly call them "Astro" here! :joy: I think in this context, it's fine just to say Cloudinary. What's important is that we make it clear these are external docs, to the site they're on, which is Astro Docs.

Author: oh i understand, that makes sense. my bad

> [Learn more about getCldOgImageUrl](https://astro.cloudinary.dev/getcldogimageurl/basic-usage).

Reviewer: ```suggestion
See [Cloudinary's `getCldOgImageUrl()` documentation](https://astro.cloudinary.dev/getcldogimageurl/basic-usage) for more information.
```

## Using Cloudinary in Node.js

For more complex asset management, uploading, or analysis, you can use the Cloudinary Node.js SDK when working in an Astro Node.js environment.

> Install the Cloudinary Node.js SDK by running:

Reviewer: ```suggestion
Install the Cloudinary Node.js SDK by running the appropriate command for your package manager:
```

<PackageManagerTabs>
  <Fragment slot="npm">
  ```shell
  npm install cloudinary
  ```
  </Fragment>
  <Fragment slot="pnpm">
  ```shell
  pnpm install cloudinary
  ```
  </Fragment>
  <Fragment slot="yarn">
  ```shell
  yarn add cloudinary
  ```
  </Fragment>
</PackageManagerTabs>

> And configure your account by running:

Reviewer: Is "running" the correct word here? This isn't a command?

Do you mean creating a file? adding environment variables?

Also, we will need a file title here using the `title=""` format in the first line of the markdown block. Having a title might have answered this question for me. 😄

Author: yeah idk why i used running here tbh 🙃 tweaked in the last commit

```js
import { v2 as cloudinary } from "cloudinary";

cloudinary.config({
  cloud_name: import.meta.env.PUBLIC_CLOUDINARY_CLOUD_NAME,
  api_key: import.meta.env.PUBLIC_CLOUDINARY_API_KEY,
  api_secret: import.meta.env.CLOUDINARY_API_SECRET,
});
```

:::note
This assumes you have your environment variables set up using the specified names.

> :::

Reviewer: We try to avoid notes/call outs as much as possible, instead including the information appropriately where it belongs.

This could be solved by telling people directly in the step above to:

- set environment variables in cloudinary however that's done (a dashboard? a file?) and show those named variables
- THEN, show this step in the Astro project, using the same names

It's fine to tell people to just use these names to keep them on the happy path, even if they could have different names. If they get to configure them on their end, then show the step of setting them here, then matching them. It just needs to be correct, even if it's opinionated.

Author: makes sense, addressed in latest commit

> Which will give you access to the Cloudinary APIs such as the Upload API, where you can upload an asset by running:

Reviewer: ```suggestion
This will give you access to all of the Cloudinary APIs to allow you to interact with your images.

For example, the Upload API allows you to upload an asset using JavaScript in your Astro project, such as in a client-side `<script>` tag.
```
Ah, I see where "running" came from! In a set of instructions, "running" is always going to mean "run this command now" not "write this function that will run a command".

So, I'm not sure if this is entirely correct, but I think something of the structure above would be a helpful way to describe this.

Author: yeah 🙈  addressed in latest
